### PR TITLE
add support for partial scanning

### PR DESCRIPTION
**Pull request by amchiclet**
Closes [SMS-545](https://linear.app/semgrep/issue/SMS-545). More context in the ticket.

This PR doesn't break semgrep's build and can be merged safely.

- [x] I ran `make setup && make` to update the generated code after editing a `.atd` file (TODO: have a CI check)
- [x] I made sure we're still backward compatible with old versions of the CLI.
      For example, the Semgrep backend need to still be able to *consume* data generated
	  by Semgrep 1.17.0.
      See https://atd.readthedocs.io/en/latest/atdgen-tutorial.html#smooth-protocol-upgrades

### CODE REVIEW

**Comment by amchiclet** [comment]
If this gets approved this week, I'll wait for the release before merging.

**Review by nmote — APPROVED** [review]

**Pull request merged**
